### DISCUSSION TITLE
Site Action Summary isn't working properly or I don't understand how to use it?

### DISCUSSION THREAD

**Discussion by morphinestyle** [akeeba/panopticon#760] [Q&A (Community Support)]
I installed Panopticon on **two servers with two completely different panels** to understand if something affected the correct functioning of Panopticon itself: one runs on Virtualmin based on nginx and one runs on CyberPanel based on OpelLitespeed.

On both **I configured a weekly "Site Action Summary"** (just to test faster, in reality I would use it every month) and I make updates on two test sites.

In theory I expect that every week I will receive the update activities (including backup?) carried out on the two test sites.

In practice I receive completely blank emails, and that's fine if I don't perform any operations on the sites; but I also get emails full of "⚙️ System Task - ℹ️ CMS update found 4.3.4 → 4.4.6" every 3 minutes, with the email "interrupted" at the last line (broken), and without the slightest indication of the update interventions made on the site.

---

Months ago I opened a discussion with the same question, but I was told that "I set SOMETHING every 3 minutes", but in reality the same thing happens even in the second installation of panopticon (new server).

Regardless: why does Panopticon fill the list of operations performed with "System Task - CMS update found" in emails, and instead doesn't report the operations carried out?
What does it mean?

**The "Site Action Summary" email should show me the operations done on the site (updates and maybe backups).**
Also: **I don't think it has any relevance to know that panopticon has checked 3,000+ times in the last week/month, every 3 minutes, that there is an update on the site.**

Am I still wrong, months later, or is something not working as it should?

**Comment by nikosdion** [comment]
The Site Action Summary sends you an excerpt of what you see in Overview, Site Reports when you filter by that site.

If you pay close attention you will see that the action reported is an attempt to run an automatic update to your site – with this second discussion it's now very clear, it wasn't before. An update which **keeps failing**. Every time the update fails, it is rescheduled. It fails again, so it gets rescheduled again. And again. And again. The time between reschedule events can vary greatly and depends on how long the failed update took to fail, how often the Joomla! Update Director task runs, and whether there are any CRON jobs running either the update or the Joomla! Update Director task. With the default settings and assuming the problem is the second issue I mention below, yeah, something around 3 to 5 minutes sounds about right.

By the way, if you have two Panopticon installations trying to update the same sites at the same time it will cause failures to be reported by one of the two installations – but not always the same one. Panopticon is not meant to be run in multiple _unrelated_ instances monitoring the same sites. I know you only did it for testing, I am trying to dissuade other people who missed the "testing" context from shooting their feet.

Also remember that by default Joomla! is using a dog slow S3 bucket to serve update downloads. This bloody thing fails more often than not, and when it doesn't it's so slow I've had faster downloads twenty years ago over a PSTN modem syncing at 48Kbps. When it does fail or becomes so bloody slow, the download of the update package can never finish (it times out) and you get into an update attempt loop. Sadly, there's no solution for this. It's a Joomla! infrastructure problem.

As I have said numerous times in different issues and discussions in this repository, this is something I reported _years_ ago to the Joomla! project. The kicker is that the Joomla! project also has a CloudFlare CDN setup which could server the updates faster and more reliably – and I told them this fact when I reported the issue. You'd think that they'd read my report and switch over to the CloudFlare CDN. You'd be wrong. They decided to **REMOVE** the CloudFlare CDN altogether from Joomla! 5.0 and later, and only use the unreliable Amazon S3 bucket (not even a Amazon CloudFront CDN distribution of the bucket which is the Amazon Web Services recommended way; straight up serving from the S3 bucket itself, like they're n00bs who have never used AWS before!). This is beyond oversight, incompetence, stupidity, or even plain old malice. It's the Joomla! maintainers being [masturbating monkeys](https://news.ycombinator.com/item?id=2541127) (in the same sense Linus Torvalds used that phrase): on the pretext of "sEcUrItY" they came up with an update delivery method which doesn't work for a great many people (including some of the _Joomla! core maintainers_, for crying out loud), thereby hurting security. You can't make that crap up…

**Comment by morphinestyle** [comment]
Now we understood, excellent! :)

I'm sorry that your suggestions were not followed... : (

You're right about the second server: I checked and forgot about the automatic patch installations, but I just turned them off (like on the first server).

**But unfortunately I remember having made numerous attempts without being able to obtain the correct list of actions.**

I'll do this:
**- Now the two servers will work in the exact same way: "If they find an update they don't have to do anything"because I only use manual updates.**
**- This week I will make small manual updates from Panopticon, both to the extensions and to the Joomla core. I will mark each update and/or backup separately.**
**- This weekend, as per Cron programming, the emails with the Action Summaries will be sent. I will show you if they actually coincide with what Panopticon did.**

**I will keep you updated!**

**P.S.: Obviously the sites are not the same!** They are two different Joomla sites, managed by two different servers where I mounted Panopticon.

**Comment by nikosdion** [comment]
I know your sites are not the same, otherwise you'd have a different problem ;) I was just commenting that for _other people_ who will read this discussion and get the wrong idea. It was 100% about preventing other people from shooting their feet.

**Comment by morphinestyle** [comment]
Here are the updates I was waiting for:

I did a test on 3 different websites, using Panopticon installed on 2 different servers, but this doesn't seem to have affected anything about the functioning of the Panopticon Site Action Summary Emails.

SITE #1: **NO updates done this week, NO backups done this week**
SITE ACTION SUMMARY EMAIL:
[
![pano1](https://github.com/user-attachments/assets/cdf311b5-421a-44a9-95c3-ae8e837df518)
](url)





SITE #2: 
06/08/2024 -- 19:32
- started a backup with Akeeba Pro from Panopticon
- updated Akeeba Backup Pro
- updated Joomla Italian Language Pack
- updated Joomla Core to 4.4.6

SITE ACTION SUMMARY EMAIL: 
![pano2](https://github.com/user-attachments/assets/19a011b0-e0d8-42b3-baae-39c1e1dfd3cc)





SITE #3 (SERVER 2): 
06/08/2024 -- 19:32
- started a backup with Akeeba Pro from Panopticon
- updated Akeeba Backup Pro
- updated JCE
- updated Joomla Core to 4.4.6

SITE ACTION SUMMARY EMAIL:
![pano3](https://github.com/user-attachments/assets/490a134a-07b3-4d69-844d-b1602bfb0245)

**Comment by morphinestyle** [comment]
I expect Panopticon to **only report backups and update activities** done on each site. Theoretically I shouldn't even see that the Panopticon Connector update was found -- I think I should only find it in the Update Summary email -- but that's the minor issue.

**Currently the emails are full of identical rows about the Core Update found, and I can't find any trace of any of the activities I've done on the sites.**

Am I doing something wrong with the Panopticon settings?

**Reply by nikosdion** [reply]
**FINALLY A SCREENSHOT**! Now it makes sense.

You have set your _site_ to report available updates to the next major version of Joomla!, by setting the Update Site to Joomla! Next. Please remember that it's your _site_ reporting available Joomla! updates, NOT Panopticon itself.

Panopticon can only install updates, or email you about available updates. Panopticon runs the system `joomlaupdatedirector` task every three minutes. This task goes through all sites which report an update to be available. When an update is available it logs the "CMS Update Found" reason before processing it.

This is what you see.

So, even though you were telling me that your site is up-to-date, it's actually not. It reports itself as **out of date**, hence the constant logging on "CMS update found". Yes, in this case we can make it so that it doesn't report this all the time.

But, see, you had me wondering for nearly two months how the bloody heck is it possible that a site that reports itself as up-to-date generating a "CMS update found" log when it was the exact opposite situation.

Well, mystery solved.

**Comment by morphinestyle** [comment]
Yes, you were right: on the #2 and #3 websites it was set to "Joomla Next".  (instead: website #1 was set to Default)

![image](https://github.com/user-attachments/assets/56163829-efad-442e-a24e-31b5087c75f9)

Currently I have set them back to "Default", so as to finally test the SITE ACTION SUMMARY email next week.

From what you say, if set to DEFAULT the problem shouldn't arise, right?
If so: will you make a patch to be able to also be set to JOOMLA NEXT?

Thank you Nicholas, i will update you on the next SITE ACTION SUMMARY email.
:)

**Reply by nikosdion** [reply]
So, it will definitely take a few hours (up to 24) for the sites to refresh their update information (remember, Joomla! caches it, by default for 6 hours) and report there are no updates. From that point on, you'll stop getting CMS update found.

**Comment by morphinestyle** [comment]
However, I still don't understand about this other site #4 (a new one, different from the first 3) because it gives me the same problem: the site has always been set to DEFAULT for Joomla updates.

**I just created a cron every 5 minutes, and the problem occurs the same, but only on WEEKLY and DAILY timeframe schedules despite having the site's Joomla updates always set to DEFAULT.**

I did a test by setting the SITE ACTION SUMMARY first to MONTHLY, then WEEKLY and finally also DAILY.

This is the result for the same website:

**MONTHLY**
![image](https://github.com/user-attachments/assets/2e3c6bf5-2f6c-43f9-bc63-37e1ff10ab8c)

**WEEKLY**
![image](https://github.com/user-attachments/assets/b2eaad5f-f5ea-4e77-9ad3-9b83101eb843)

**DAILY**
![image](https://github.com/user-attachments/assets/beac937a-bd2d-4f34-9690-99c66c31d1dd)

**Reply by nikosdion** [reply]
https://github.com/akeeba/panopticon/commit/7abebd325a6580f60b5fee84c18d97e70f0db7fb

**Reply by morphinestyle** [reply]
WordPress websites has the same problem Nicholas

**Reply by nikosdion** [reply]
https://github.com/akeeba/panopticon/commit/982e607106a429d684c96a8549abe149b5efde19
